Sources/Persistence/PersistencePerformanceMetricsTracker.swift

```diff
@@ -0,0 +1,159 @@
+import Foundation
+import Result
+
+public protocol PersistencePerformanceMetricsTracker: PerformanceMetricsTracker {
+
+    /// A closure to be used when stopping measuring memory reads/writes.
+    typealias MemoryAccessStopClosure<E: Error> = (_ result: Result<(blobSize: UInt64, memorySize: UInt64), E>) -> Void
+
+    /// A closure to be used when stopping measuring disk reads/writes.
+    typealias DiskAccessStopClosure<E: Error> = (_ result: Result<(blobSize: UInt64, diskSize: UInt64), E>) -> Void
+
+    /// The metadata key used for the used memory.
+    var usedMemoryMetadataKey: PerformanceMetrics.Metadata.Key { get }
+
+    /// The metadata key used for the used disk.
+    var usedDiskMetadataKey: PerformanceMetrics.Metadata.Key { get }
+
+    /// The metadata key used for the blob size being read/written.
+    var blobSizeMetadataKey: PerformanceMetrics.Metadata.Key { get }
+
+    /// The metadata key used for the error.
+    var errorMetadataKey: PerformanceMetrics.Metadata.Key { get }
+
+    /// The metric identifier for a memory read operation.
+    var memoryReadIdentifier: PerformanceMetricsTracker.Identifier { get }
```

Comment:
You can use only `Identifier` instead of `PerformanceMetricsTracker.Identifier`.